netty/src/test/java/io/grpc/transport/netty/NettyClientTransportTest.java

```diff
@@ -0,0 +1,263 @@
+    MethodDescriptor<String, String> method = MethodDescriptor.create(MethodType.UNARY,
+            "/testService/test", 10, TimeUnit.SECONDS, StringMarshaller.INSTANCE,
+            StringMarshaller.INSTANCE);
+    for (NettyClientTransport transport : transports) {
+      SettableFuture rpcFuture = SettableFuture.create();
+      rpcFutures.add(rpcFuture);
+      ClientStream stream = transport.newStream(method, new Metadata.Headers(),
+              new TestClientStreamListener(rpcFuture));
+      stream.request(1);
+      stream.writeMessage(messageStream());
+      stream.halfClose();
+    }
+
+    // Wait for the RPCs to complete.
+    for (SettableFuture rpcFuture : rpcFutures) {
+      rpcFuture.get(10, TimeUnit.SECONDS);
+    }
+  }
+
+  private NioEventLoopGroup newGroup() {
+    NioEventLoopGroup group = new NioEventLoopGroup(1);
+    groups.add(group);
+    return group;
+  }
+
+  private static InputStream messageStream() {
+    return new ByteArrayInputStream(MESSAGE.getBytes());
+  }
+
+  private static class TestClientStreamListener implements ClientStreamListener {
+    private final SettableFuture<?> future;
+
+    TestClientStreamListener(SettableFuture<?> future) {
+      this.future = future;
+    }
+
+    @Override
+    public void headersRead(Metadata.Headers headers) {
+    }
+
+    @Override
+    public void closed(Status status, Metadata.Trailers trailers) {
+      if (status.isOk()) {
+        future.set(null);
+      } else {
+        future.setException(status.asException());
+      }
+    }
+
+    @Override
+    public void messageRead(InputStream message) {
+    }
+
+    @Override
+    public void onReady() {
+    }
+  }
+
+  private static class TestServerListener implements ServerListener {
+
+    @Override
+    public ServerTransportListener transportCreated(final ServerTransport transport) {
+      return new ServerTransportListener() {
+
+        @Override
+        public ServerStreamListener streamCreated(final ServerStream stream, String method,
+                                                  Metadata.Headers headers) {
+          stream.request(1);
+          return new ServerStreamListener() {
+
+            @Override
+            public void messageRead(InputStream message) {
+              // Just echo back the message.
+              stream.writeMessage(messageStream());
+            }
+
+            @Override
+            public void onReady() {
+            }
+
+            @Override
+            public void halfClosed() {
+              // Just close when the client closes.
+              stream.close(Status.OK, new Metadata.Trailers());
+            }
+
+            @Override
+            public void closed(Status status) {
+            }
+          };
+        }
+
+        @Override
+        public void transportTerminated() {
+        }
+      };
+    }
+
+    @Override
+    public void serverShutdown() {
+    }
+  }
+
+  private static class StringMarshaller implements Marshaller<String> {
```

Comment:
Not sure I follow ... that seems to implement the `AsciiMarshaller` interface, not `Marshaller`.